src/GraphQLParser/Visitors/PrintContextExtensions.cs

```diff
@@ -15,27 +15,34 @@ public static class PrintContextExtensions
     /// using <see cref="CancellationToken"/> from it.
     /// </summary>
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    public static ValueTask WriteAsync<TContext>(this TContext context, ROM value)
+    public static async ValueTask WriteAsync<TContext>(this TContext context, ROM value)
         where TContext : IPrintContext
     {
+        if (value.Length == 0)
+            return;
+
+        context.NewLinePrinted = value.Span[value.Length - 1] == '\n';
```

Comment:
That approach is much easier than trying to calculate the flag in each place where `WriteAsync` is called.